packages/amplify_datastore/android/src/main/kotlin/com/amazonaws/amplify/amplify_datastore/AmplifyDataStorePlugin.kt

```diff
@@ -770,4 +831,77 @@ class AmplifyDataStorePlugin : FlutterPlugin, MethodCallHandler {
             }
         }
     }
+
+    override fun addAuthPlugin(callback: (kotlin.Result<Unit>) -> Unit) {
+        try {
+            Amplify.addPlugin(NativeAuthPluginWrapper { nativeAuthPlugin })
+            LOG.info("Added Auth plugin")
+            callback(kotlin.Result.success(Unit))
+        } catch (e: Exception) {
+            LOG.error(e.message)
+            callback(kotlin.Result.failure(e))
+        }
+    }
+
+    override fun updateCurrentUser(user: NativeAuthUser?) {
+        currentUser = if (user != null) {
+            AuthUser(user.userId, user.username)
+        } else {
+            null
+        }
+    }
+
+    override fun addApiPlugin(
+        authProvidersList: List<String>,
+        callback: (kotlin.Result<Unit>) -> Unit
+    ) {
+        try {
+            val authProviders = authProvidersList.map { AuthorizationType.valueOf(it) }
+            if (flutterAuthProviders == null) {
+                flutterAuthProviders = FlutterAuthProviders(authProviders, nativeApiPlugin!!)
+            }
+            Amplify.addPlugin(
+                AWSApiPlugin
+                    .builder()
+                    .apiAuthProviders(flutterAuthProviders!!.factory)
+                    .build()
+            )
+            LOG.info("Added API plugin")
+            callback(kotlin.Result.success(Unit))
+        } catch (e: Exception) {
+            LOG.error(e.message)
+            callback(kotlin.Result.failure(e))
+        }
+    }
+
+    override fun configure(
+        version: String,
+        config: String,
+        callback: (kotlin.Result<Unit>) -> Unit
+    ) {
+        coroutineScope.launch(dispatcher) {
+            try {
+                val configuration = AmplifyConfiguration.builder(JSONObject(config))
+                    .addPlatform(UserAgent.Platform.FLUTTER, version)
+                    .devMenuEnabled(false)
+                    .build()
+                Amplify.configure(configuration, context)
+                withContext(Dispatchers.Main) {
+                    callback(kotlin.Result.success(Unit))
+                }
+            } catch (e: Amplify.AlreadyConfiguredException) {
+                val flutterError = FlutterError(
+                    "AmplifyAlreadyConfiguredException",
+                    e.toString()
+                )
+                callback(kotlin.Result.failure(flutterError))
+            } catch (e: AmplifyException) {
+                val flutterError = FlutterError(
+                    "postExceptionToFlutterChannel",
```

Comment:
I think this was a bad copy/paste 😅 . It should be "AmplifyException". I think I copied the method name instead of the error code. Thanks for catching it. Updated.